### ISSUE TITLE
LocalResource spawn_local called from outside of a task::LocalSet

### ISSUE THREAD

**Issue by VVishion** [leptos-rs/leptos#2717]
**Describe the bug**

```rust
async fn create_list() -> Vec<String> {
    vec![String::from("Ehm")]
}

#[component]
pub fn App() -> impl IntoView {
    //let list = Resource::new(|| (), |_| async { create_list().await });
    let list = LocalResource::new(|| async { create_list().await });

    view! {
        <div id="app">
            {move || list.get()}
        </div>
    }
}
```
Panics on the server with 
```
thread 'main' panicked at /home/vvishion/.cargo/git/checkouts/leptos-2c8932a5258b3db6/4bc999e/any_spawner/src/lib.rs:136:17:
`spawn_local` called from outside of a `task::LocalSet`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The above code doesnt panic with `let list = Resource::new(|| (), move |_| async move { create_list().await });`

**Leptos Dependencies**
```toml
leptos = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_axum = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_meta = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_router = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
``` 
at the time of commit https://github.com/leptos-rs/leptos/commit/4bc999ead36ab9eeb2b80e13ef945b343e9e3393

**To Reproduce**
compile

**Expected behavior**
Working as `Resource`

**Additional context**
I can provide the full workspace if this isnt enough to reproduce

**Comment by gbj** [comment]
Thanks. Should be fixed by 798e567, this was introduced by the switch to spawning a local task for local resources, which should never have spawned a task (even just a `pending()` task) on the server anyway.

Note that there are two additional issues that will come up with the minimal reproduction:
1. You'll get a panic with the error message that reading a `LocalResource` on the server outside `Suspense` is not possible — this is because in order to still support non-nullable access with `.await`, `LocalResource` is defined as always pending on the server, then short-circuits Suspense. So rather than hanging indefinitely we panic if it's outside Suspense.
2. This particular example then runs into an edge case: because it is not actually async, the browser assumes that the data are ready and have already been server rendered, so the Suspense never resolves on the client. Making the async function actually async (as it would be in a real example) solves the issue.

So I am consider this fixed for the adapted MRE:
```rust
#[server]
async fn create_list() -> Result<Vec<String>, ServerFnError> {
    Ok(vec![String::from("Ehm")])
}

#[component]
pub fn App() -> impl IntoView {
    //let list = Resource::new(|| (), |_| async { create_list().await });
    let list = LocalResource::new(|| async { create_list().await });

    view! {
        <Suspense>
            <div id="app">{move || format!("{:?}", list.get())}</div>
        </Suspense>
    }
}
```

**Comment by VVishion** [comment]
I can confirm the true async case is fixed.
